river/SeatStatus.zig

```diff
@@ -78,3 +87,22 @@ pub fn sendMode(self: Self, mode: [*:0]const u8) void {
         self.seat_status.sendMode(mode);
     }
 }
+
+pub fn sendKeyboardLayout(self: Self, device: *InputDevice, opt_layout: ?[*:0]const u8) void {
+    if (self.seat_status.getVersion() < 5) return;
+
+    const identifier = std.fmt.allocPrint(
+        util.gpa,
+        "{s}\x00",
+        .{device.identifier},
+    ) catch return;
+    defer util.gpa.free(identifier);
+
+    const c_ident = @ptrCast([*:0]const u8, identifier.ptr);
```

Comment:
Should I change `InputDevice.identifier` to be a null terminated sting to avoid this?

Reply:
Sure, that would be more idiomatic. Also note that allocPrintZ() can be used instead of this pointer cast.